bpf/process/types/sockaddr.h

```diff
@@ -13,34 +13,52 @@ struct sockaddr_in_type {
 	__u64 sin_addr[2];
 };
 
-/* set_event_from_sockaddr_in(event, address)
+struct sockaddr_un_type {
+	__u16 sin_family;
+	char sun_path[108];
+};
+
+struct sockaddr_event {
+	__u16 family;
+	union {
+		struct sockaddr_in_type in;
+		struct sockaddr_un_type un;
+	};
+};
+
+/* set_event_from_sockaddr(event, address)
  *
  * Populate the event args with the sock info.
  */
 FUNC_INLINE void
-set_event_from_sockaddr_in(struct sockaddr_in_type *event, struct sockaddr *address)
+set_event_from_sockaddr(struct sockaddr_event *event, struct sockaddr *address)
 {
 	struct sockaddr_in6 *ipv6 = (struct sockaddr_in6 *)address;
 	struct sockaddr_in *ipv4 = (struct sockaddr_in *)address;
+	struct sockaddr_un *un = (struct sockaddr_un *)address;
 	__u32 addr;
 
 	memset(event, 0, sizeof(*event));
-	if (probe_read(&event->sin_family, sizeof(event->sin_family), _(&address->sa_family)) < 0)
+	if (probe_read(&event->family, sizeof(event->family), _(&address->sa_family)) < 0)
 		return;
-	switch (event->sin_family) {
+	switch (event->family) {
 	case AF_INET:
 		// Read the 32 bit address into temporary var and then copy so we don't have to
 		// consider endianness and alignment.
 		probe_read(&addr, sizeof(addr), _(&ipv4->sin_addr));
-		event->sin_addr[0] = addr;
-		probe_read(&event->sin_port, sizeof(event->sin_port), _(&ipv4->sin_port));
+		event->in.sin_addr[0] = addr;
+		probe_read(&event->in.sin_port, sizeof(event->in.sin_port), _(&ipv4->sin_port));
 		break;
 	case AF_INET6:
-		probe_read(&event->sin_addr, sizeof(event->sin_addr), _(&ipv6->sin6_addr));
-		probe_read(&event->sin_port, sizeof(event->sin_port), _(&ipv6->sin6_port));
+		probe_read(&event->in.sin_addr, sizeof(event->in.sin_addr), _(&ipv6->sin6_addr));
+		probe_read(&event->in.sin_port, sizeof(event->in.sin_port), _(&ipv6->sin6_port));
+		break;
+	case AF_UNIX:
+		probe_read_str(event->un.sun_path, sizeof(event->un.sun_path), un->sun_path);
```

Comment:
This doesn't feel safe as sockaddrs are often user-controlled, and the sun_path might be 108 chars long and not null-terminated (if I've understood correctly).

Also, maybe we should support abstract sockets where the first byte is 0x00, and path name can contain null bytes?